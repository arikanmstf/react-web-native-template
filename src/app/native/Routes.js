import React from 'react';
import { MemoryRouter as Router, Switch, Route } from 'react-router-native';
import { HOME, DUMMY_TWO } from 'constants/routes/routeNames';

// Screens
import DummyScreen from 'screens/dummy/DummyNativeContainer';
import DummyScreenTwo from 'screens/dummyTwo/DummyNativeContainer';

const Routes = () => (
  <Router>
    <Switch>
      <Route exact path={HOME} component={DummyScreen} />
      <Route exact path={DUMMY_TWO} component={DummyScreenTwo} />
    </Switch>
  </Router>
);

export default Routes;
