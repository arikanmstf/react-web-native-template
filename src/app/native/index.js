require('react-native');
/* eslint-disable */
function interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

interopRequireDefault(require('react'));

const expo = interopRequireDefault(require('expo'));

const App = interopRequireDefault(require('./App.js'));

if (process.env.NODE_ENV === 'development') {
  expo.default.KeepAwake.activate();
}

expo.default.registerRootComponent(App.default);
/* eslint-enable */
