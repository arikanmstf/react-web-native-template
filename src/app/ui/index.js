export { default as Text } from './Text/Web';
export { default as Button } from './Button/Web';
export { default as View } from './View/Web';
