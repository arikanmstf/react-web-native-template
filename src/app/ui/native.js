export { default as Text } from './Text/Native';
export { default as Button } from './Button/Native';
export { default as View } from './View/Native';
export { default as Image } from './Image/Native';
