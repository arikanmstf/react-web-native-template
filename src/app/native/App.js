// @flow
import React from 'react';

// Redux
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import reducers from 'store/reducers';

// Theme
import { ThemeProvider } from 'react-native-material-ui';
import * as COLOR from 'constants/theme/color';

// Routes
import Routes from './Routes';

if (typeof global.self === 'undefined') {
  global.self = global;
}

const createStoreWithMiddleware = applyMiddleware(thunk)(createStore);
const uiTheme = {
  palette: {
    primaryColor: COLOR.green500,
  },
};

const App = () => {
  return (
    <Provider store={createStoreWithMiddleware(reducers)}>
      <ThemeProvider uiTheme={uiTheme}>
        <Routes />
      </ThemeProvider>
    </Provider>
  );
};

export default App;
