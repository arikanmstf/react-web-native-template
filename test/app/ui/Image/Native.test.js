import React from 'react';
import { shallow } from 'enzyme';
import { Image } from 'ui/native';

describe('src/app/ui/Image/Native.js', () => {
  const props = {
    source: {
      uri: 'dummyUri',
    },
  };

  it('Render: props: { source: { uri } }', () => {
    const wrapper = shallow(<Image {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Render: props: { alt }', () => {
    const newProps = {
      ...props,
      alt: 'dummyAlt',
    };

    const wrapper = shallow(<Image {...newProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Render: props: { style }', () => {
    const newProps = {
      ...props,
      style: 'dummyStyle',
    };

    const wrapper = shallow(<Image {...newProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
