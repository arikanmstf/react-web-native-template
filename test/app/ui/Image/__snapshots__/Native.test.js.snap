// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/app/ui/Image/Native.js Render: props: { alt } 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Image
    alt="dummyAlt"
    source={
      Object {
        "uri": "dummyUri",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "alt": "dummyAlt",
      "source": Object {
        "uri": "dummyUri",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "alt": "dummyAlt",
        "source": Object {
          "uri": "dummyUri",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`src/app/ui/Image/Native.js Render: props: { source: { uri } } 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Image
    source={
      Object {
        "uri": "dummyUri",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "source": Object {
        "uri": "dummyUri",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "source": Object {
          "uri": "dummyUri",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`src/app/ui/Image/Native.js Render: props: { style } 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Image
    source={
      Object {
        "uri": "dummyUri",
      }
    }
    style="dummyStyle"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "source": Object {
        "uri": "dummyUri",
      },
      "style": "dummyStyle",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "source": Object {
          "uri": "dummyUri",
        },
        "style": "dummyStyle",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
